[DRAFT] Alpha-Omega Report June 2024 - Rust Foundation (#8)
JoelMarcey/alpha-omega: alpha/engagements/2024/Rust Foundation/update-2024-06.md

## Engineering

### Exposed crates.io tokens

Using a local copy of crates.io, Walter has a process running [gitleaks](https://gitleaks.io/) regex scripts to detect exposed credentials. Nine exposed tokens were found and working with Adam through a security incident process, the tokens were removed and the authors associated with those tokens have been notified.

### Security infrastructure to the cloud

Walter is leading the porting of our security initiative infrastructure from basically local machines into the cloud, specifically Google Cloud Platform (GCP). There is now an active crates mirror where security scans are regularly run with tools like [Painter](https://github.com/rustfoundation/Painter) and [Typomania](https://github.com/rustfoundation/typomania). The results of these scans are stored in a neo4J database and elastic search can be used to find specific results.

### Crate deletion RFC

> Tobias wrote and published a crate deletion [RFC](https://github.com/rust-lang/rfcs/pull/3660). This RFC provides crate owners and administrators functionality a mechanism to delete  crates from crates.io under certain [conditions](https://github.com/Turbo87/rust-rfcs/blob/crates-io/crate-deletions/text/3660-crates-io-crate-deletions.md#proposal). The RFC is in discussion, but has a general positive response.

Reviewer: ```suggestion
Tobias wrote and published a crate deletion [RFC](https://github.com/rust-lang/rfcs/pull/3660). This RFC provides crate owners with a mechanism to delete crates from crates.io under certain [conditions](https://github.com/Turbo87/rust-rfcs/blob/crates-io/crate-deletions/text/3660-crates-io-crate-deletions.md#proposal). The RFC is in discussion, but has a general positive response so far.
```

### Crates.io RSS Feed

An RSS feed is being added to crates.io. The [PR](https://github.com/rust-lang/crates.io/pull/8908) creates an RSS feed published at https://static.crates.io/rss/updates.xml. The feed is synced with the database in a background job after every successful publish of a new version. It includes the latest 100 published versions with the crate name, version number, crate description, URL and publish date. The feed is created via https://github.com/rust-syndication/rss.

## Announcements and Community

### Safety Critical Consortium

The Foundation [announced](https://foundation.rust-lang.org/news/announcing-the-safety-critical-rust-consortium/) a Rust Safety Critical Consortium with nine founding members to focus on encouraging Rust in safety critical applications. Interest in the consortium is abundant and overwhelming, with over twenty more wanting to join or collaborate, and more coming every day. At RustConf, the consortium plans to define a charter and goals, along with a mechanism and process to efficiently add new members to the consortium.

### Rust/C++ Interop Engineer Lead on Staff

> [Jon Bauman](https://foundation.rust-lang.org/news/welcoming-rust-c-interoperability-engineer-jon-bauman-to-the-rust-foundation-team/) is our new software engineer lead helping manage our all-important, Google funded [Rust/C++ interop initiative](https://foundation.rust-lang.org/news/welcoming-rust-c-interoperability-engineer-jon-bauman-to-the-rust-foundation-team/). In his first month in the role, Jon has met with more than a dozen people with a vested interest or expertise in interop, and is starting to plan a proposed interop strategy.

Reviewer 2: Depending on how you want to parse "met" and whether one includes only Foundation-external stakeholders this number could be as low as 10 or as high as 18, but I doubt anybody's going to come asking for receipts.

Author: Thanks. Yeah, I didn't know the exact number and "dozen" felt about right.

### RustSec Advisory Team

> Walter a member of the RustSec Advisory team to review advisories.

Reviewer 3: ```suggestion
Walter has joined the RustSec Advisory team to help review advisories.
```
